adb/device.go

```diff
@@ -165,13 +161,14 @@ func (d *Device) Upgrade(path string) error {
 	return d.install([]string{"-r"}, path)
 }
 
-func getResultLine(out io.Reader) string {
-	scanner := bufio.NewScanner(out)
-	for scanner.Scan() {
-		l := scanner.Text()
-		if strings.HasPrefix(l, "Failure") || strings.HasPrefix(l, "Success") {
-			return l
-		}
+func getResultLine(output []byte) string {
+	str := strings.TrimRight(string(output[:]), "\n")
+	if strings.HasPrefix(str, "Success") {
+		return str
+	}
+	fail := strings.Index(str, "Failure")
+	if fail >= 0 {
+		return str[fail:]
```

Comment:
Why this index of Failure? If you do what I suggested above regarding reading line by line, you can just use `HasPrefix` like before.